GraknConsole.java

```diff
@@ -233,22 +248,30 @@ private void runQuery(GraknClient.Transaction tx, String queryString) {
     }
 
     private <T> void printCancellableResult(Stream<T> results, Consumer<T> printFn) {
-        long counter = 0;
+        AtomicLong counter = new AtomicLong();
         Instant start = Instant.now();
-
         try {
-            boolean[] isCancelled = new boolean[1];
-            terminal.handle(Terminal.Signal.INT, s -> isCancelled[0] = true);
             Iterator<T> iterator = results.iterator();
-            while (!isCancelled[0] && iterator.hasNext()) {
-                counter++;
-                printFn.accept(iterator.next());
-            }
+            Future<?> answerPrintingJob = executorService.submit(() -> {
+                while (iterator.hasNext() && !Thread.interrupted()) {
+                    printFn.accept(iterator.next());
+                    counter.getAndIncrement();
+                }
+            });
+            terminal.handle(Terminal.Signal.INT, s -> answerPrintingJob.cancel(true));
+            answerPrintingJob.get();
+            Instant end = Instant.now();
+            printer.info("answers: " + counter + ", duration: " + Duration.between(start, end).toMillis() +" ms");
+        } catch (InterruptedException | ExecutionException e) {
+            e.printStackTrace();
+        } catch (CancellationException e) {
+            Instant end = Instant.now();
+            printer.info("answers: " + counter + ", duration: " + Duration.between(start, end).toMillis() +" ms");
+
+            printer.info("The stream was shut down in the console, but the Grakn Server might still be working on the last answers for some time.");
```

Comment:
In this print statement, the wordings you use are not straightforward enough for the user to understand, and also the sentence can be written in a better way.

Try to imagine the user's perspective: he's executing a query, and he's pressing CTRL+C to cancel it. Use words and sentences that can be understood from someone from that perspective.